src/commands/replay-sqs-dlq.js

```diff
@@ -1,42 +1,122 @@
 const _ = require("lodash");
 const { getAWSSDK } = require("../lib/aws");
 const { getQueueUrl } = require("../lib/sqs");
+const { getTopicArn } = require("../lib/sns");
 const { Command, flags } = require("@oclif/command");
 const { checkVersion } = require("../lib/version-check");
 
 class ReplaySqsDlqCommand extends Command {
 	async run() {
 		const { flags } = this.parse(ReplaySqsDlqCommand);
-		const { dlqQueueName, queueName, region, concurrency, profile, keep } = flags;
+		const {
+			dlqQueueName,
+			targetName,
+			targetType,
+			region,
+			concurrency,
+			profile,
+			keep
+		} = flags;
 
 		global.region = region;
 		global.profile = profile;
 		global.keep = keep;
 
 		checkVersion();
 
-		this.log(`finding the queue [${dlqQueueName}] in [${region}]`);
+		this.log(`finding the SQS DLQ [${dlqQueueName}] in [${region}]`);
 		const dlqQueueUrl = await getQueueUrl(dlqQueueName);
 
-		this.log(`finding the queue [${queueName}] in [${region}]`);
-		const queueUrl = await getQueueUrl(queueName);
+		let sendToTarget;
+		switch (targetType) {
+			case "SNS":
+				this.log(`finding the SNS topic [${targetName}] in [${region}]`);
+				sendToTarget = this.sendToSNS(await getTopicArn(targetName));
+				break;
+			case "Kinesis":
+				sendToTarget = this.sendToKinesis(targetName);
+				break;
+			default:
+				this.log(`finding the SQS queue [${targetName}] in [${region}]`);
+				sendToTarget = this.sendToSQS(await getQueueUrl(targetName));
+				break;
+		}
 
 		this.log(
-			`replaying events from [${dlqQueueUrl}] to [${queueUrl}] with ${concurrency} concurrent pollers`
+			`replaying events from [${dlqQueueUrl}] to [${targetType}:${targetName}] with ${concurrency} concurrent pollers`
 		);
-		this.replay(dlqQueueUrl, queueUrl, concurrency);
+		await this.replay(dlqQueueUrl, concurrency, sendToTarget);
 
 		this.log("all done!");
 	}
 
-	async replay(dlqQueueUrl, queueUrl, concurrency) {
+	async replay(dlqQueueUrl, concurrency, sendToTarget) {
 		const promises = _.range(0, concurrency).map(() =>
-			this.runPoller(dlqQueueUrl, queueUrl)
+			this.runPoller(dlqQueueUrl, sendToTarget)
 		);
 		await Promise.all(promises);
 	}
 
-	async runPoller(dlqQueueUrl, queueUrl) {
+	sendToSQS(queueUrl) {
+		const AWS = getAWSSDK();
+		const SQS = new AWS.SQS();
+
+		return async messages => {
+			const sendEntries = messages.map(msg => ({
+				Id: msg.MessageId,
+				MessageBody: msg.Body,
+				MessageAttributes: msg.MessageAttributes
+			}));
+			await SQS.sendMessageBatch({
```

Comment:
What happens when you have more than 10? 10 is the maximum per batch. I don't see it in the docs, but we've experienced this error in prod - https://docs.aws.amazon.com/AWSSimpleQueueService/latest/APIReference/API_SendMessageBatch.html

Reply:
10 is the max size per batch for both reads and writes, so since we're processing each batch we receive so we're good